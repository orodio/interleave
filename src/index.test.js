import { interleave } from './index'

const s = arr => `[${arr.join(',')}]`

const snap = (a, b) => {
  // const r = interleave(a,b)
  it(`interleave(${s(a)},${s(b)})`, () => {
    expect(interleave(a,b)).toMatchSnapshot()
  })
}

describe('interleave', () => {
  snap([],          [])
  snap([1],         [])
  snap([],          [1])
  snap([1],         [2])
  snap([1,2],       [3])
  snap([1,2],       [3,4])
  snap([1,2,3,4,5], ["a", "b", "c", "d"])
  snap([1,2,3,4,5], ["a", "b", "c", "d", "e"])
  snap([1,2,3,4,5], ["a", "b", "c", "d", "e", "f"])
})
