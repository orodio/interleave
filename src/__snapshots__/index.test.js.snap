// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interleave interleave([],[]) -> [] 1`] = `Array []`;

exports[`interleave interleave([],[1]) -> [] 1`] = `Array []`;

exports[`interleave interleave([1,2,3,4,5],[a,b,c,d,e,f]) -> [1,a,2,b,3,c,4,d,5,e] 1`] = `
Array [
  1,
  "a",
  2,
  "b",
  3,
  "c",
  4,
  "d",
  5,
  "e",
]
`;

exports[`interleave interleave([1,2,3,4,5],[a,b,c,d,e]) -> [1,a,2,b,3,c,4,d,5,e] 1`] = `
Array [
  1,
  "a",
  2,
  "b",
  3,
  "c",
  4,
  "d",
  5,
  "e",
]
`;

exports[`interleave interleave([1,2,3,4,5],[a,b,c,d]) -> [1,a,2,b,3,c,4,d,5] 1`] = `
Array [
  1,
  "a",
  2,
  "b",
  3,
  "c",
  4,
  "d",
  5,
]
`;

exports[`interleave interleave([1,2],[3,4]) -> [1,3,2,4] 1`] = `
Array [
  1,
  3,
  2,
  4,
]
`;

exports[`interleave interleave([1,2],[3]) -> [1,3,2] 1`] = `
Array [
  1,
  3,
  2,
]
`;

exports[`interleave interleave([1],[]) -> [1] 1`] = `
Array [
  1,
]
`;

exports[`interleave interleave([1],[2]) -> [1,2] 1`] = `
Array [
  1,
  2,
]
`;
